mopidy_youtube/apis/youtube_bs4api.py

```diff
@@ -52,21 +70,21 @@ def run_search(cls, query):
                 "div", class_=["yt-lockup-video", "yt-lockup-playlist"]
             )
 
-            if not results:
-                logger.info("nothing in the soup, trying japi")
-                json_regex = r'window\["ytInitialData"] = ({.*?});'
-                extracted_json = re.search(json_regex, result.text).group(1)
-                result_json = json.loads(extracted_json)["contents"][
-                    "twoColumnSearchResultsRenderer"
-                ]["primaryContents"]["sectionListRenderer"]["contents"][0][
-                    "itemSectionRenderer"
-                ][
-                    "contents"
-                ]
-                return jAPI.json_to_items(cls, result_json)
-            else:
+            if results:
                 return cls.soup_to_items(cls, results)
 
+            logger.info("nothing in the soup, trying japi")
+
+            yt_data = cls._find_yt_data(result.text)
+            json = yt_data["contents"]\
```

Comment:
Calling this `json` breaks things; better to call it result_json or similar, as previously, I think.